foam_controlwrapper/foam_mesh.py

```diff
@@ -0,0 +1,772 @@
+""" Mesh module
+
+This module contains the implementation to store, access,
+and modify a mesh and related data
+
+"""
+import uuid
+from simphony.cuds.abstractmesh import ABCMesh
+from simphony.core.cuba import CUBA
+from simphony.cuds.mesh import Point, Face, Cell
+import simphony.core.data_container as dc
+from foam_files import FoamFiles
+import simphonyfoaminterface as foamface
+from foam_templates import head
+from foam_templates import scalarTemplates, vectorTemplates
+import tempfile
+import os
+import numpy
+
+
+class FoamMesh(ABCMesh):
+    """ Proxy class to communicate with OpenFoam engines mesh data.
+
+    Parameters
+    ----------
+    name : str
+        name of mesh
+
+    mesh : ABCMesh
+       mesh to store
+
+    Attributes
+    ----------
+    name : str
+        name of mesh
+    data : DataContainer
+        DataContainer to store mesh global data (midpoint, ...).
+    _uuidToFoamLabel : dictionary
+        Mapping from uuid to OpenFoam label number
+    _foamCellLabelToUuid : dictionary
+        Mapping from OpenFoam cell label number to uuid
+    _foamFaceLabelToUuid : dictionary
+        Mapping from  OpenFoam face label number to uuid
+    _foamEdgeLabelToUuid : dictionary
+        Mapping from  OpenFoam edge label number to uuid
+    _foamPointLabelToUuid : dictionary
+        Mapping from  OpenFoam point label number to uuid
+
+    """
+
+    def __init__(self, name, mesh=None, path=None):
+        self.name = name
+        self.data = dc.DataContainer()
+        self.time = str(0)
```

Comment:
> For example, what should print code

> for cell in mesh_inside_wrapper.iter_cells():
> print cell.data[CUBA.PRESSURE]

The high-level objects provide the **current** state of the modeling engine. So, in your example, the printed values of pressure would correspond to whatever the current time step is. 

For example,  the following prints the results at time steps 0, N and 2N:


```
...
wrapper = openfoam.FoamControlWrapper()
...
wrapper.SP[CUBA.NUMBER_OF_TIME_STEPS] = N
....
mesh_inside_wrapper = wrapper.add_mesh(mesh_from_file)
....
# prints values at timestep 0
for cell in mesh_inside_wrapper.iter_cells():
print cell.data[CUBA.PRESSURE]

wrapper.run()

# prints values at timestep N
for cell in mesh_inside_wrapper.iter_cells():
print cell.data[CUBA.PRESSURE]

wrapper.run()

# prints values at timestep 2N
for cell in mesh_inside_wrapper.iter_cells():
print cell.data[CUBA.PRESSURE]
```

In the current API, the wrapper does not provide historical data.  However our API does provide the user with a way to get historical data by querying the wrapper at different stages that the user is interested in (like in the above example).